src/addrman.cpp

```diff
@@ -521,3 +533,84 @@ void CAddrMan::SetServices_(const CService& addr, ServiceFlags nServices)
 int CAddrMan::RandomInt(int nMax){
     return GetRandInt(nMax);
 }
+
+void CAddrMan::ResolveCollisions_()
+{
+    for (std::set<int>::iterator it = m_tried_collisions.begin(); it != m_tried_collisions.end();) {
+        int id_new = *it;
+
+        bool erase_collision = false;
+
+        // If id_new not found in mapInfo remove it from m_tried_collisions
+        if (mapInfo.count(id_new) != 1) {
+            erase_collision = true;
+        } else {
+            CAddrInfo& info_new = mapInfo[id_new];
+
+            // Which tried bucket to move the entry to.
+            int tried_bucket = info_new.GetTriedBucket(nKey);
+            int tried_bucket_pos = info_new.GetBucketPosition(nKey, false, tried_bucket);
+
+            if (!info_new.IsValid()) { // id_new may no longer map to a valid address
+                erase_collision = true;
+
+            // The position in the tried bucket is not empty
+            } else if (vvTried[tried_bucket][tried_bucket_pos] != -1) {
```

Comment:
Nit: The above if case looks misleadingly like a single-no-curling-brace-block (I misread it as such) until you realize the thing below the comment (`// The position...`) is actually ending the block.

Maybe move the comment inside and indented inside the else block?

Reply:
It looked weird below the else if and in a brief search I couldn't find any other examples of that style in code. Instead I moved it to right of the if else.

Reply:
There are plenty of examples of this, I think.

E.g. https://github.com/bitcoin/bitcoin/blob/bf3353de90598f08a68d966c50b57ceaeb5b5d96/src/validation.cpp#L1399-L1402

https://github.com/bitcoin/bitcoin/blob/bf3353de90598f08a68d966c50b57ceaeb5b5d96/src/validation.cpp#L2416-L2418